### PR TITLE
ci(pinner): Create `issues-pinner.yml` workflow

### LINKED ISSUE #7055: feat: Create a workflow that label/unlabel issues on pin/unpin
To enhace with #7053 stale labeling bypass, create a GitHub Actions workflow that automanage labels on pin/unpin issues.

When
- Pin an issue... add a label `📌 pinned`
- Unpin an issue... remove the label `📌 pinned`
- Optional: close a issue... remove the label `📌 pinned`

### PR DESCRIPTION
**Pull request by davorpa**
### Description

A workflow to monitor pin/unpin issue event types and add/remove state labels accordingly

Needs declare a new label: ~`📌 pinned`~  `:pushpin: pinned`

Resolves #7055

Workflow runs:
	- Tested with davorpa/free-programming-books#16 issue
	- https://github.com/davorpa/free-programming-books/actions/runs/2983767547
	- https://github.com/davorpa/free-programming-books/actions/runs/2983768434
	- https://github.com/davorpa/free-programming-books/actions/runs/2983769061
	- https://github.com/davorpa/free-programming-books/actions/runs/2983769167

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: .github/workflows/issues-pinner.yml [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+# This allows a subsequently queued workflow run to interrupt/wait for previous runs
+concurrency:
+#  group: '${{ github.workflow }} @ ${{ github.event.issue.number || github.run_id }} :: ${{ github.event.action }}'
+#  cancel-in-progress: true
+  group: '${{ github.workflow }} @ ${{ github.event.issue.number || github.run_id }}'
+  cancel-in-progress: false
+
+jobs:
+
+  if-pinned:
```
Comment (LuigiImVector) [inline .github/workflows/issues-pinner.yml:23]: The code looks good to me, I would, however, remove the superfluous newlines below

Comment (davorpa) [reply .github/workflows/issues-pinner.yml:23]: > I would, however, remove the superfluous newlines below

Do you refer to compact sourcecode?

Comment (LuigiImVector) [reply .github/workflows/issues-pinner.yml:23]: like
```
jobs:
  if-pinned:
    name: On pinning an issue
    if: github.event.action == 'pinned'
    permissions:
      issues: write
    runs-on: ubuntu-latest
    steps:
...
```

Comment (davorpa) [reply .github/workflows/issues-pinner.yml:22]: > The code looks good to me, I would, however, remove the superfluous newlines below

Done. See commit cf3b28a730f4e56c7a48ee11af12e824c4aae4d8

**Review by LuigiImVector — APPROVED** [review]

**Comment by eshellman** [comment]
do you know of another repo where this is done?

**Comment by eshellman** [comment]
any comments, @mohitsaxenaknoldus ?

**Comment by davorpa** [comment]
> do you know of another repo where this is done?

Not directly. But workflow is easy to newbies. I apply some merged knowledges:

- Base docs: [`actions-ecosystem/action-add-labels`](https://github.com/actions-ecosystem/action-add-labels) & [`actions-ecosystem/action-remove-labels`](https://github.com/actions-ecosystem/action-remove-labels) & [`actions-ecosystem/action-size`](https://github.com/actions-ecosystem/action-size)
- Run jobs concurrently to take performance into account. Stack stragegy is used (FIFO), so new runs waits for those older
- Security. Scoping permissions to what really needed. In general `issues: read` and no git/filesystem (`content: none`) access.
- Supercharging workflows with summaries. https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/
- It only affect to pin/unpin issue events image not PRs/commits

**Comment by davorpa** [comment]
A job cleanup is made. Instead have defined 3 jobs we have 3 steps. It seems go a bit faster

**Comment by eshellman** [comment]
please add a header comment in the code that describes the purpose of the action and exactly what it does.

**Comment by davorpa** [comment]
> please add a header comment in the code that describes the purpose of the action and exactly what it does.

This action adds a label to an issue when is pinned and removes it when unpinned.

The label is enhanced with stale.yml workflow. Pinned issues never stales due to the label is declared as exemption

**Comment by eshellman** [comment]
> 
> This workflow adds a label to an issue when is pinned and removes it when unpinned.
> 
> The label is enhanced with stale.yml workflow. Pinned issues never stales due to the label is declared as exemption

that helps. now add it to the file

**Comment by davorpa** [comment]
> that helps. now add it to the file

Done

**Review by LuigiImVector — APPROVED** [review]

**Comment by LuigiImVector** [comment]
Nice work

**Pull request merged**